src/metrics/prometheus.go

```diff
@@ -0,0 +1,206 @@
+// +build prometheus
+
+// Package metrics contains support for reporting metrics to an external server,
+// currently a Prometheus pushgateway. Because plz runs as a transient process
+// we can't wait around for Prometheus to call us, we've got to push to them.
+package metrics
+
+import (
+	"os/user"
+	"runtime"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/push"
+	"gopkg.in/op/go-logging.v1"
+
+	"core"
+)
+
+var log = logging.MustGetLogger("metrics")
+
+const maxErrors = 3
```

Comment:
nit: A little documentation that says please will stop trying to push metrics for the build after it has failed this many times?